docs/topics/proc_selecting-the-jdk-builder-image-for-your-application.adoc

```diff
@@ -0,0 +1,57 @@
+[id='selecting-the-jdk-builder-image-for-your-application_{context}']
+= Selecting a JDK image for your application
+Edit your application `pom.xml` file to specify the image for building and running your {runtime} application.
+
+.Prerequisites
+* A `pom.xml` file for your application.
+* Maven installed.
+
+.Procedure
+. Open the `pom.xml` file in the application root directory and locate the `fabric8.generator.from` property.
+
+. Replace the `fabric8.generator.from` property to specify the version of RHEL and OpenJDK you intend to use.
+
+RHEL 7 with OpenJKD 8::
++
+[source,xml,options="nowrap",subs="attributes+"]
+----
+<fabric8.generator.from>{name-2si-image-openjdk8-rhel7}</fabric8.generator.from>
```

Comment:
A thought, not necessarily criticism: It may be worth considering putting this information in the reference module and reformat that as a list instead of a table.

Reply:
Not entirely happy with the modules but I'm not sure your suggestion helps (me) :-) So simple in DITA :-)

Reply:
Right now, the flow of the procedure is interrupted with what is essentially a reference table in step 2. I think it would be easier to follow to put that reference table in a separate module and put a requirement on having selected the value from the table in prerequisites.

Reply:
@tradej I don't wholly disagree but I feel that sending the reader to another module with no clear navigation back is not a great alternative. I made the second step less intrusive so let me know what you think.

Reply:
@huntley As Adam and Lada are suggesting, the content in this procedure makes sense as part of the _Preparing..._ module, so if you read _Deploying a Vert.x application to OpenShift_ from top to bottom, you read:

1. the assembly introduction (which can explain that you will need the URL for the POM file), then 
1. the table (that would also contain the URLs) in the _OpenJDK images for Red Hat..._ reference module,
1. in _Preparing...,_ you read a mention of the URL (that you're already aware of) in the prerequisites and step 2 only says to input the URL.

This way, the reader doesn't have to backtrack at all.

The independence of the individual modules is not violated because the _OpenJDK images for Red Hat..._ module can easily be replaced with any other module that contains URLs to input in the POM file.

An added benefit that I've just realised is that when you get another combination in the support matrix (a new version of JDK or some such), you only have to update one module.

In any case, I maintain that I don't think your solution is bad or that it must be changed, I just believe there's a more elegant way of going about it.